In [80]:

# Lo primero siempre es importar la biblioteca y en este caso se le asigna un nombre, np, para no tener que escribir numpy cada vez
import numpy as np
# Creación de vectores (arrays para numpy) de dos dimensiones.
# Observar que se crean a partir de las listas de Python
A = np.array([[1, 2, 3], [4, 5, 6]])
print(A)
# Observar que el resultado es una matriz en este caso de 2 filas y 3 columnas

[[1 2 3]
 [4 5 6]]


In [1]:
import numpy as np

# Crear un array de NumPy
arr = np.array([1, 2, 3, 4, 5])

# Reversar el array
reversed_arr = arr[::-1]

#arr[a:b:c]
#a inicio
#b fin
#c paso

In [81]:
# Crear un array numpy mediante uso de la función arange de numpy.
#arange es un arreglo unidimensional con valores igualmente espaciados dentro de un intervalo especificado
# Consultar que hace la función 'arange' y qué parámetros recibe.
import numpy as np
arr = np.arange(0, 1, 0.2)
print(arr)

[0.  0.2 0.4 0.6 0.8]


In [82]:
# Crear un array de numpy mediante la función linespace de numpy
# subdividir una lista en intervalos iguales

#np.linspace(start, end, subdivisiones)

import numpy as np
arr = np.linspace(0, 2*np.pi, 4)
print(arr)

[0.         2.0943951  4.1887902  6.28318531]


In [83]:
#Crear ceros y unos en numpy
import numpy as np

#zero = np.zeros((filas, columnas))
a= np.zeros((2, 3))
print(a)
print()
#zero = np.ones((filas, columnas))
b= np.ones((2, 2))
print(b)

[[0. 0. 0.]
 [0. 0. 0.]]

[[1. 1.]
 [1. 1.]]


In [84]:
#Asignación de arrays en numpy:  observar que al modificar uno se modifica también el original.
import numpy as np
A = np.zeros((2, 2))
C = A
C[0, 0] = 1
print(C)
print(A)

[[1. 0.]
 [0. 0.]]
[[1. 0.]
 [0. 0.]]


In [85]:
#multiplicar y sumar arrays

import numpy as np
A = np.array([[2, 2],[3,4]])
B = np.array([[2, 2],[3,4]])


print(A)
print(B)
print(A*B)
#los arrays no son matrices, ellos se multiplican multiplicando sus coordenadas iguales

print(A+B)


#cuando le das a un array una suma algebarica de un float o int lo hace paara cada elemnto de la lista
#


[[2 2]
 [3 4]]
[[2 2]
 [3 4]]
[[ 4  4]
 [ 9 16]]
[[4 4]
 [6 8]]


In [86]:
#Arreglo identidad = Matriz identidad
import numpy as np
#el comando es eye o identity
#np.identity(n) o tambien np.eye(n)
print(np.eye(3))

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]


In [87]:
#Arreglos random

import numpy as np
print(np.random.random((2,3)))
#tambien sirve np.random.randn((fila,columna))


[[0.87089745 0.00551274 0.26093466]
 [0.33593257 0.28616176 0.27442586]]


In [10]:
#cambiar la forma de arreglos en numpy
import numpy as np
arr=np.array([2,3,4,5,6,7])
#sigue la forma arr.reshape((fila, columna))
print(arr.reshape((2,3)))

my_array = np.array([[1,2,3],
                     [4,5,6]])
print (np.transpose(my_array))
# numpy.transpose It will not affect the original array, but it will create a new array.
#Output
#[[1 4]
# [2 5]
# [3 6]]

my_array = np.array([[1,2,3],
                        [4,5,6]])
print (my_array.flatten())

#Output
#[1 2 3 4 5 6]

#The tool flatten creates a copy of the input array flattened to one dimension.



#en una lista podemos usar arr.max()/arr.min() para encontrar el maximo de una lista 
print(arr.max())


[[2 3 4]
 [5 6 7]]
[[1 4]
 [2 5]
 [3 6]]
[1 2 3 4 5 6]
7


In [89]:
import numpy as np

arr=np.array([[2,3,4,5,6]])


raiz = np.sqrt(arr)

##Devuelve la raiz para cada elemento del arreglo

exponente = np.exp(arr)

#Devuelve e ^ elemento para cada elemento del arreglo

seno=np.sin(arr)

#Calcula el seno trigonométrico de cada valor en el arreglo

coseno=np.cos(arr)

#Calcula el coseno trigonométrico de cada valor en el arreglo

logaritmo=np.log(arr)

#Calcula el logaritmo en base diez de cada valor en el arreglo

print(raiz)
print(exponente)
print(seno)
print(coseno)
print(logaritmo)


[[1.41421356 1.73205081 2.         2.23606798 2.44948974]]
[[  7.3890561   20.08553692  54.59815003 148.4131591  403.42879349]]
[[ 0.90929743  0.14112001 -0.7568025  -0.95892427 -0.2794155 ]]
[[-0.41614684 -0.9899925  -0.65364362  0.28366219  0.96017029]]
[[0.69314718 1.09861229 1.38629436 1.60943791 1.79175947]]


In [90]:
#Redondeando decimales
import numpy as np
arr=np.array([0.69292946, 0.9365295 , 0.65682359, 0.72770856, 0.83268616])
print(np.round(arr, 2))
#redondea cada valor del array a dos decimales



# arr[0] devuelve 0.69292946 

#arr[:] = 0  Devuelve array([0., 0., 0., 0., 0.])

#arr[2:5] = 0.5  Devuelve array([0. , 0. , 0.5, 0.5, 0.5])



[0.69 0.94 0.66 0.73 0.83]


In [91]:
array_to_copy = np.array([1, 2, 3])

copied_array = array_to_copy.copy()

print(array_to_copy)
print(copied_array)

#ahora

copied_array[0] = 9

print(copied_array)

print(array_to_copy)
print()

#se usa el metodo .copy para tener un nuevo array porque si usas igualas un objeto al mismo array y reescribes el objeto, se reescribe el array original
#ejemplo:

new_array = np.array([6, 7, 8, 9])

second_new_array = new_array[0:2]

print(second_new_array)

second_new_array[1] = 4

print(second_new_array) 

#Devuelve array([6, 4]), como se esperaba

print(new_array) 

#Devuelve array([6, 4, 8, 9]) 


[1 2 3]
[1 2 3]
[9 2 3]
[1 2 3]

[6 7]
[6 4]
[6 4 8 9]


In [92]:
#a.ndim : Devuelve el número de dimensiones del array a.

#a.shape : Devuelve una tupla con las dimensiones del array a.

#a.size : Devuelve el número de elementos del array a.

#a.dtype: Devuelve el tipo de datos de los elementos del array a.

#a.floor: Redondea para abajo
#a.ceil: Redondea para arriba
#a.ceil: Redondea para arriba
#a.rint: Redondea usualmente

In [93]:
a = np.array([[1, 2, 3], [4, 5, 6]])
print(a[1, 0])  # Acceso al elemento de la fila 1 columna 0

print(a[1][0])  # Otra forma de acceder al mismo elemento

print(a[:, 0:2])
#[[1 2]
# [4 5]]


#Para filtar elementos de un array se usa a(condicion):

print(a[(a % 2 == 0)])
#[2 4 6] Fijese que crea un nuevo array con los que cumplen esa condicion
print(a[(a % 2 == 0) &  (a > 2)])
#[4 6]

4
4
[[1 2]
 [4 5]]
[2 4 6]
[4 6]


In [3]:
import numpy
my_array = numpy.array([[2, 5], 
                        [3, 7],
                        [1, 3],
                        [4, 0]])

print (numpy.min(my_array, axis = 0))   #minimo en cada columna      #Output : [1 0]
print (numpy.min(my_array, axis = 1))   #minimo en cada fila      #Output : [2 3 1 0]
print (numpy.min(my_array, axis = None))      #Output : 0
print (numpy.min(my_array))                   #Output : 0


my_array = numpy.array([ [1, 2], [3, 4] ])

print numpy.mean(my_array, axis = 0)        #Output : [ 2.  3.]
print numpy.mean(my_array, axis = 1)        #Output : [ 1.5  3.5]
print numpy.mean(my_array, axis = None)     #Output : 2.5
print numpy.mean(my_array)                  #Output : 2.5
#The mean tool computes the arithmetic mean along the specified axis.


my_array = numpy.array([ [1, 2], [3, 4] ])

print numpy.var(my_array, axis = 0)         #Output : [ 1.  1.]
print numpy.var(my_array, axis = 1)         #Output : [ 0.25  0.25]
print numpy.var(my_array, axis = None)      #Output : 1.25
print numpy.var(my_array)                   #Output : 1.25
#The var tool computes the arithmetic variance along the specified axis.

my_array = numpy.array([ [1, 2], [3, 4] ])

print numpy.std(my_array, axis = 0)         #Output : [ 1.  1.]
print numpy.std(my_array, axis = 1)         #Output : [ 0.5  0.5]
print numpy.std(my_array, axis = None)      #Output : 1.11803398875
print numpy.std(my_array)                   #Output : 1.11803398875

#The std tool computes the arithmetic standard deviation along the specified axis.


[1 0]
[2 3 1 0]
0
0


In [ ]:
int_array = str_array.astype(int)
#toma un array y redefine sus elemntos, en eset caso los convierte en enteros




In [11]:
import numpy

array_1 = numpy.array([1,2,3])
array_2 = numpy.array([4,5,6])
array_3 = numpy.array([7,8,9])

print (numpy.concatenate((array_1, array_2, array_3))) 
#el output es una tupla
#Output
#[1 2 3 4 5 6 7 8 9]


array_4 = numpy.array([[1,2,3],[0,2,2]])
array_5 = numpy.array([[0,0,0],[7,8,9]])

print (numpy.concatenate((array_4, array_5), axis = 1))   

#Output
#[[1 2 3 0 0 0]
# [0 2 2 7 8 9]] 

print (numpy.concatenate((array_4, array_5), axis = 0))   
#Output
#[[1 2 3]
# [0 2 2]
# [0 0 0]
# [7 8 9]]

[1 2 3 4 5 6 7 8 9]
[[1 2 3 0 0 0]
 [0 2 2 7 8 9]]
[[1 2 3]
 [0 2 2]
 [0 0 0]
 [7 8 9]]


In [1]:
import numpy

print(numpy.poly([-1, 1, 1, 10]))        #Output : [  1 -11   9  11 -10]
#the coefficients of a polynomial with the given sequence of roots.

print(numpy.roots([1, 0, -1]))           #Output : [-1.  1.]
#the roots of a polynomial with the given coefficients

print(numpy.polyint([1, 1, 1]))          #Output : [ 0.33333333  0.5         1.          0.        ]
#an antiderivative (indefinite integral) of a polynomial.

print(numpy.polyder([1, 1, 1, 1]))       #Output : [3 2 1]
#the derivative of the specified order of a polynomial.

print(numpy.polyval([1, -2, 0, 2], 4))   #Output : 34
# evaluates the polynomial at specific value.

print(numpy.polyfit([0,1,-1, 2, -2], [0,1,1, 4, 4], 2))
#Output : [  1.00000000e+00   0.00000000e+00  -3.97205465e-16]
#np.polyfit((coords1),(coords2),grado)

#polyadd, polysub, polymul, and polydiv





[  1. -11.   9.  11. -10.]


# Algebra Lineal

In [3]:
import numpy as np

#Multiplicacion de Vectores
a = np.array([1, 2, 3])
b = np.array([1, 0, 1])

print(a @ b)
print(a.dot(b))
#ambas formas de multiplicar el vector son equivalentes


A = numpy.array([ 1, 2 ])
B = numpy.array([ 3, 4 ])

print(numpy.cross(A, B))     #Output : -2
#esto es el prouct cruzado de dos vectores

c = np.array([3, 4])
print(np.linalg.norm(a))
#el modulo de un vector



d = np.array([[1, 2, 3], [4, 5, 6]])
e = np.array([[1, 1], [2, 2], [3, 3]])
print(d @ e)
#[[14 14]
#[32 32]]
#multiplicacion de matrices

print(d.T) #matriz traspuesta
print(d.trace()) #tarza de matriz


print(np.linalg.det(d @ e)) #determinante de matriz

f = np.array([[1, 2], [3, 4]])
print(np.linalg.inv(f)) #inversa de matriz

#Av=λv
#A es una matriz, λ sería un valor propio, y v un vector propio
vals, vecs = numpy.linalg.eig([[1 , 2], [2, 1]])
print(vals)                                      #Output : [ 3. -1.]
print (vecs)                                      #Output : [[ 0.70710678 -0.70710678]
                                                #          [ 0.70710678  0.70710678]]




# Sistema de dos ecuaciones y dos incógnitas
# x + 2y = 1
# 3x + 5y = 2 
matriz_coeficientes = np.array([[1, 2], [3, 5]])
matriz_resultado = np.array([1, 2])

print(np.linalg.solve(matriz_coeficientes, matriz_resultado))
#[-1.  1.]

4
4
-2
3.7416573867739413
[[14 14]
 [32 32]]
[[1 4]
 [2 5]
 [3 6]]
6
0.0
[[-2.   1. ]
 [ 1.5 -0.5]]
[ 3. -1.]
[[ 0.70710678 -0.70710678]
 [ 0.70710678  0.70710678]]
[-1.  1.]


C:\Users\cesar\AppData\Local\Temp\ipykernel_9624\3258532673.py:15: DeprecationWarning: Arrays of 2-dimensional vectors are deprecated. Use arrays of 3-dimensional vectors instead. (deprecated in NumPy 2.0)
  print(numpy.cross(A, B))     #Output : -2


Comprobación de Resultados

Podemos verificar que los resultados son correctos al comprobar que Av=λv 
    Para λ=3
    A[0.7071 0.7071]≈3[0.7071 0.7071]
 

    Para λ=−1λ=−1:
    A[−0.7071 0.7071]≈−1[−0.7071 0.7071]
